storybook/qmlTests/tests/tst_SwapModal.qml

```diff
@@ -1815,5 +1815,51 @@ Item {
                 compare(amountToSendInput.input.text, expectedAmount)
             }
         }
+
+        function test_no_auto_refresh_when_proposalLoading_or_approvalPending() {
+            fetchSuggestedRoutesCalled.clear()
+            root.swapFormData.fromTokenAmount = "0.0001"
+            root.swapFormData.selectedAccountAddress = "0x7F47C2e18a4BBf5487E6fb082eC2D9Ab0E6d7240"
+            root.swapFormData.selectedNetworkChainId = 11155111
+            root.swapFormData.fromTokensKey = "ETH"
+            // for testing making it 1.2 seconds so as to not make tests running too long
+            root.swapFormData.autoRefreshTime = 1200
+
+            // Launch popup
+            launchAndVerfyModal()
+
+            // check if fetchSuggestedRoutes called
+            fetchSuggestedRoutesCalled.wait()
+            compare(fetchSuggestedRoutesCalled.count, 1)
+
+            // no new calls to fetch new proposal should be made as the proposal is still loading
+            wait(root.swapFormData.autoRefreshTime*2)
+            compare(fetchSuggestedRoutesCalled.count, 1)
+
+            // emit routes ready
+            let txHasRouteApproval = root.dummySwapTransactionRoutes.txHasRoutesApprovalNeeded
+            txHasRouteApproval.uuid = root.swapAdaptor.uuid
+            root.swapStore.suggestedRoutesReady(txHasRouteApproval, "", "")
+
+            // now refresh can occur as no propsal or signing is pending
+            fetchSuggestedRoutesCalled.wait()
+            compare(fetchSuggestedRoutesCalled.count, 2)
+
+            // emit routes ready
+            txHasRouteApproval.uuid = root.swapAdaptor.uuid
+            root.swapStore.suggestedRoutesReady(txHasRouteApproval, "", "")
+
+            verify(root.swapAdaptor.swapOutputData.approvalNeeded)
+            verify(!root.swapAdaptor.approvalPending)
+
+            // sign approval and check that auto refresh doesnt occur
+            root.swapAdaptor.sendApproveTx()
+
+            // no new calls to fetch new proposal should be made as the approval is pending
+            verify(root.swapAdaptor.swapOutputData.approvalNeeded)
+            verify(root.swapAdaptor.approvalPending)
+            wait(root.swapFormData.autoRefreshTime*2)
```

Comment:
This is even more fishy, there should be no `wait()` calls _at all_

Reply:
Here I need the wait cause I am waiting 2 * time for auto refresh to make sure that auto refresh doesn't occur in case a old request is already pending. If you know a better way to check signal not emitted please help out. try compare wont work as it can check even before a certain time and the statement can return true even though auto refresh occured.

Reply:
Hmm, in theory, if the signals hasn't been fired, the respective SignalSpy's `count` should be 0

Reply:
yes but if I call I don't get the necessity to call wait() on the SignalSpy; it should be possible to just do:

tryCompare(fetchSuggestedRoutesCalled, "count", 0) and before this I clear the spy.

it will hit 0 initially and this condition will pass even though it maybe be called a few ms after.

Reply:
Ah I get it, it's initially `0` so the condition is immediately satisfied... :thinking: 

Ok, I think then just keep it the way it is

Reply:
yes, I replace it all the other places where I can though now. Here I think it is needed :)

Reply:
Thought: Maybe the `wait` calls indicate we should eventually externalise the refresh logic. Probably to a `RoutingProvider` component and the SwapModal just to display the data as received.

WDYT, would it be an improvement if we externalise this logic in a future task?

Reply:
I guess you mentioned this in another PR as well, you have created another component that already does this for community right?

Reply:
Hmm, right! I think we already have a task to take a look on it, right?

Reply:
https://github.com/status-im/status-desktop/issues/15502

Reply:
Nice, thanks! Sorry, didn't mean to sound like a broken record 😄 Forgot about it